core/src/main/scala/com/yahoo/maha/core/RequestModel.scala

```diff
@@ -1115,6 +1119,40 @@ object RequestModel extends Logging {
     }
   }
 
+  def validateLengthForFilterValue(publicTable: PublicTable, filter: Filter): Boolean = {
+    val dataType = {
+      publicTable match {
+        case publicDim: PublicDimension => publicDim.nameToDataTypeMap(publicDim.columnsByAliasMap(filter.field).name)
+        case publicFact: PublicFact => publicFact.dataTypeForAlias(publicFact.columnsByAliasMap(filter.field).alias)
+        case _ => None
+      }
+    }
+
+    dataType match {
+      case None => throw new IllegalArgumentException(s"Unable to find expected PublicTable as PublicFact or PublicDimension.")
+      case StrType(length, _, _) => filter match {
+        case InFilter(_, values, _, _) =>
+          if (length == 0 && values.forall(_.length <= MAX_ALLOWED_STR_LEN) || values.forall(_.length <= length)) true else false
+        case NotInFilter(_, values, _, _) =>
+          if (length == 0 && values.forall(_.length <= MAX_ALLOWED_STR_LEN) || values.forall(_.length <= length)) true else false
+        case EqualityFilter(_, value, _, _) =>
+          if (length == 0 && value.length <= MAX_ALLOWED_STR_LEN || value.length <= length) true else false
+        case NotEqualToFilter(_, value, _, _) =>
+          if (length == 0 && value.length <= MAX_ALLOWED_STR_LEN || value.length <= length) true else false
+        case LikeFilter(_, value, _, _) =>
+          if (length == 0 && value.length <= MAX_ALLOWED_STR_LEN || value.length <= length) true else false
+        case BetweenFilter(_, from, to) =>
+          if (from.length <= MAX_ALLOWED_STR_LEN && to.length <= MAX_ALLOWED_STR_LEN) true else false
+        case IsNullFilter(_, _, _) | IsNotNullFilter(_, _, _) | PushDownFilter(_) | OuterFilter(_) | OrFliter(_) => true
+        case _ => throw new Exception(s"Unhandled FilterOperation.")
+      }
+      case _ => filter match {
```

Comment:
this code is redundant, we dont want to check the length on other data types, int/ doubles are already checked by the json parser. you can remove this case _ and always return true in this case